cee_syslog_handler/__init__.py

```diff
@@ -271,3 +271,39 @@ def filter(self, record):
         """
         found = self._pattern.search(record.getMessage())
         return not found
+
+
+class RegexRedactFilter(logging.Filter):
+    """
+    This filter redacts parts of log messages and exception traces
+    by substituting through a provided regular expression.
+    Use with caution: with great power comes great responsibility
+    """
+    def __init__(self, filter_regex=None, replace_string="<redacted>"):
+        super(RegexRedactFilter, self).__init__()
+        self._pattern = re.compile(filter_regex)
+        self._replacement = replace_string
+
+    def redact(self, string):
+        return re.sub(self._pattern, self._replacement, string)
+
+    def filter(self, record):
+        if isinstance(record.msg, str):
+            record.msg = self.redact(record.msg)
+        elif isinstance(record.msg, BaseException):
+            # redact exception messages that are thrown
+            record.msg.args = tuple(self.redact(arg) for arg in record.msg.args)
+        else:
+            # potentially impacts other filters
+            # takes str conversion from getMessage to here
+            record.msg = self.redact(str(record.msg))
+
+        # also redact args, which are merged with msg in getMessage
+        # through msg % args
+        if isinstance(record.args, dict):
+            for k in record.args.keys():
+                record.args[k] = self.redact(record.args[k])
+        else:
+            record.args = tuple(self.redact(arg) for arg in record.args)
```

Comment:
What happens if one of the args is an integer or float?

Reply:
Well spotted. The redact function currently would fail on those and those should just stay unchanged. At least that would be sufficient for our purposes.